lib/utils/display.js

```diff
@@ -12,6 +50,57 @@ class Display {
     log.pause()
   }
 
+  static clean (output) {
+    if (typeof output === 'string') {
+      // Strings are cleaned inline
+      return stripC01(output)
+    }
+    if (!output || typeof output !== 'object') {
+      // Numbers, booleans, null all end up here and don't need cleaning
+      return output
+    }
+    // output && typeof output === 'object'
+    // We can't use hasOwn et al for detecting the original but we can use it
+    // for detecting the properties we set via defineProperty
+    if (
+      output[inspect.custom] &&
+      (!Object.hasOwn(output, originalCustomInspect))
+    ) {
+      // Save the old one if we didn't already do it.
+      Object.defineProperty(output, originalCustomInspect, {
+        value: output[inspect.custom],
+        writable: true,
+      })
+    }
+    if (!Object.hasOwn(output, originalCustomInspect)) {
+      // Put a dummy one in for when we run multiple times on the same object
+      Object.defineProperty(output, originalCustomInspect, {
+        value: function () {
+          return this
+        },
+        writable: true,
+      })
+    }
+    // Set the custom inspect to our own function
+    Object.defineProperty(output, inspect.custom, {
```

Comment:
In the future, if we were to have a single shared interface for displaying to the terminal would we still need to define `inspect.custom` as a property on every object passed through `npm.dispay`?

I understand the reason for it in this PR and agree this is the best way to handle it. My question isn't a block on this PR but I want to understand if we think this code is an opportunity to influence a new display interface if/when we get there.

Reply:
The reason for this is because we ALSO pass these through to our logging interface, which we have less control over what we can manipulate in flight.